### PR TITLE
KTOR-583 Allow setting custom session for Darwin engine

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-583](https://youtrack.jetbrains.com/issue/KTOR-583/IosClientEngine-URLSession-access)

### CODE REVIEW

**Comment by e5l** [comment]
@andreyaksenov could you check KDocs?

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt [hunk 1]
```diff
@@ -54,6 +54,12 @@ public class DarwinClientEngineConfig : HttpClientEngineConfig() {
     public var challengeHandler: ChallengeHandler? = null
         private set
 
+    /**
+     * Session to use for making HTTP requests.
```
Comment (andreyaksenov) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt:58]: > Specifies a session used to make HTTP requests.

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt [hunk 2]
```diff
@@ -80,6 +86,13 @@ public class DarwinClientEngineConfig : HttpClientEngineConfig() {
         }
     }
 
+    /**
+     * Set [session] to be used to make HTTP requests, [null] to create default session.
```
Comment (andreyaksenov) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt:90]: > Sets a [session] ...

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt [hunk 3]
```diff
@@ -80,6 +86,13 @@ public class DarwinClientEngineConfig : HttpClientEngineConfig() {
...
+     * If the preconfigured session is set, [configureSession] block will be ignored.
+     */
+    public fun usePreconfiguredSession(session: NSURLSession? = null) {
+    }
```
Comment (rsinukov) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt:94]: should you set variable here?

Comment (rsinukov) [reply ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt:94]: please add test

**Review by rsinukov — APPROVED** [review]

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt (hunk 3 above)
Comment (e5l) [reply ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinClientEngineConfig.kt:94]: oh, thanks. Will do

**Pull request merged**